src/Symfony/Component/DependencyInjection/Loader/YamlFileLoader.php

```diff
@@ -239,6 +239,9 @@ private function parseDefaults(array &$content, $file)
      */
     private function parseDefinition($id, $service, $file, array $defaults)
     {
+        if (preg_match('/^_[a-zA-Z0-9_]*$/', $id)) {
+            @trigger_error(sprintf('Service names that start with an underscore are deprecated since Symfony 3.3 and will be reserved in 4.0. Rename the "%s" service or define it in XML instead.', $id), E_USER_DEPRECATED);
```

Comment:
I don't think it's a good idea that the allowed service ids depend on the config format. In the past, we've worked hard to make all formats work the same in spite of their differences (for example, by adding PHP constants support for Yaml files). If we want to deprecate something, we could deprecate just the `_defaults` string as service id (whatever the config format used).